docs/book/beyond_basics/webhook_installer_generator.md

```diff
@@ -0,0 +1,148 @@
+
+The secret contains
+- the serving certificate and its corresponding private key
+- the signing CA certificate and its corresponding private key
+
+Webhook installer can be very helpful for
+- faster iteration during development
+- easier for deploying in production if policy allows
+
+Webhook installer is on by default. Set
+[`DisableWebhookConfigInstaller`](https://github.com/kubernetes-sigs/controller-runtime/blob/a8ea2056444a5d74d7408e4b8798cbe63b14066b/pkg/webhook/server.go#L57-L60)
+to true to turn it off.
+
+## Webhook Manifests Generator
+
+From cluster administrators perspective, they may want to have the webhook configurations
+installation to be a separate process from running the webhook server,
+since permissions to create and update webhook configurations are considered as high privileges.
+
+Similar to other generators in [controller-tools](https://github.com/kubernetes-sigs/controller-tools),
+webhook manifest generator is annotation-driven.
+
+How the webhook manifest generator works
+1) It parses the annotations to configure
+[Webhook](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook/admission#Webhook).
+1) It parses the annotations to configure
+[ServerOptions](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook#ServerOptions).
+1) It uses the library in controller-runtime, which is the same machinery as
+the installer, to generate the webhook configuration manifests.
+
+#### Comment Group
+
+A comment group represents a sequence of comments with no empty lines between.
+Comment group is a concept that is important for writing and parsing annotations correctly.
+
+For example, the following comments are in one comment group
+
+```
+// +kubebuilder:webhook:groups=apps
+// +kubebuilder:webhook:resources=deployments
+```
+
+The following comments are in 2 comments groups
+
+```
+// +kubebuilder:webhook:groups=apps
+
+// +kubebuilder:webhook:resources=deployments
+```
+
+#### Annotations
+
+Each comment line that starts with `+kubebuilder:webhook:` will be processed to extract annotations.
+
+The annotations can be grouped in 2 categories based on what struct they configure.
+
+The _first_ category is for each individual webhook.
+They are used to set the fields in [`Webhook`](https://github.com/kubernetes-sigs/controller-runtime/blob/a8ea2056444a5d74d7408e4b8798cbe63b14066b/pkg/webhook/admission/webhook.go#L55-L80) struct.
+The annotations for the same webhook are allowed to span across multiple lines as long as they are prefixed with
+`+kubebuilder:webhook:` and in the __same__ comment group.
+
+For example, the following is for one single webhook
+
+```
+// +kubebuilder:webhook:groups=apps,versions=v1,resources=deployments,verbs=create,update
+// +kubebuilder:webhook:name=mutating-create-update-deployment.testproject.org
+// +kubebuilder:webhook:path=/mutating-create-update-deployment
+// +kubebuilder:webhook:type=mutating,failure-policy=fail
```

Comment:
I'd like to see some explanation or an example showing which files these annotations should go into.